packages/core/src/components/autosuggest/autosuggest.tsx

```diff
@@ -195,13 +201,31 @@ export class Autosuggest {
       this.input.addEventListener("input", this.onInput);
       this.input.addEventListener("keydown", this.onKeyDown);
       this.input.addEventListener("focusin", this.onFocusIn);
+
+      window.addEventListener("resize", this.onWindowResize);
     });
   }
 
   disconnectedCallback() {
     this.input?.removeEventListener("input", this.onInput);
     this.input?.removeEventListener("keydown", this.onKeyDown);
     this.input?.removeEventListener("focusin", this.onFocusIn);
+
+    window.removeEventListener("resize", this.onWindowResize);
+  }
+
+  private onWindowResize = debounce(() => this.setListboxContainerMaxBlockSize(), 150);
+
+  private setListboxContainerMaxBlockSize(): void {
+    const availableBlockSize = window.innerHeight - this.host.getBoundingClientRect().bottom;
+
+    if (this.listboxContainer && this.showSuggestions && availableBlockSize > listItemBlockSize) {
+      if (availableBlockSize < listboxMaxBlockSize) {
+        this.listboxContainer.style.maxBlockSize = availableBlockSize - 2 * listboxPaddingBlock + "px";
+      } else {
+        this.listboxContainer.style.maxBlockSize = listboxMaxBlockSize + "px";
+      }
```

Comment:
Imperatief / handmatig eigenschappen van elementen wijzigen is een antipatroon. Beter is om de waarden in een lokale state te bewaren en vanuit de `render()` functie in JSX (declaratief) te zetten.